src/FlatpickrComponent.jsx

```diff
@@ -21,12 +21,13 @@ class FlatpickrComponent extends React.Component {
 	*/
 	componentDidMount() {
 		const options = {
-			...this.props.opts,
 			onChange: this.onChange,
 			altInput: true,
 			altFormat: 'D M d, Y', // TODO localize
 			defaultDate: this.props.value
 		};
+
+		Object.assign(options, this.props.opts);
```

Comment:
why this change? The object spread operator is effectively the same as `Object.assign` but it enforces immutability.

If you move the `...this.props.opts` down to the bottom of the `const options` object, it will correctly override the defaults.